runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/fn/control/TimerReceiver.java

```diff
@@ -119,7 +119,8 @@ public boolean receive(String pCollectionId, Object receivedElement) {
         String timerId = timerSpec.timerId();
 
         TimerInternals timerInternals = stepContext.namespacedToUser().timerInternals();
-        timerInternals.setTimer(namespace, timerId, timer.getTimestamp(), timeDomain);
+        timerInternals.setTimer(
+            namespace, timerId, timer.getTimestamp(), timer.getTimestamp(), timeDomain);
```

Comment:
the second timer.getTimestamp() should be timer.getOutputTimestamp()

Reply:
Here timer is not belongs to TimerInternals.java so its don`t have getOutputTimestamp() method.

Reply:
So we should add outputTimestamp to TimerInternals.

Reply:
We need to make sure that the Timer object always gets constructed with an output timestamp so that we can plumb it through here.